udf/mrn_udf_highlight_html.cpp

```diff
@@ -415,12 +445,69 @@ MRN_API char *mroonga_highlight_html(UDF_INIT *init,
   *is_null = 0;
   GRN_BULK_REWIND(&(info->result));
 
-  if (!highlight_html(ctx,
-                      reinterpret_cast<grn_pat *>(keywords),
-                      args->args[0],
-                      args->lengths[0],
-                      &(info->result))) {
-    goto error;
+  if (info->target_type == MRN_TARGET_TYPE_HTML) {
+    const char *previous_position = args->args[0];
+    const char *end_position = args->args[0] + args->lengths[0];
+    const char *current_position = args->args[0];
+    bool in_tag = false;
+
+    while (current_position < end_position) {
+      int char_length = grn_charlen(ctx, current_position, end_position);
+
+      if (char_length == 0) {
+        goto error;
+      }
+      
+      current_position += char_length;
+
+      if (*current_position == '<') {
```

Comment:
This still assumes the current character is a single byte character.
We need to check the current character byte size is `1` before we access the current character.

Reply:
Fixed.